ckanext/harvest/logic/action/get.py

```diff
@@ -124,9 +126,7 @@ def harvest_source_list(context, data_dict):
     TODO: Use package search
     '''
 
-    check_access('harvest_source_list', context, data_dict)
-
-    sources = _get_sources_for_user(context, data_dict)
+    sources = _get_sources_for_user(context, data_dict, harvest_source_list=True)
```

Comment:
Rather than pass in a flag for where the call is coming from, perhaps pass in the organisation_id, as it appears that the other call which uses `_get_sources_for_user` is called from the command line so potentially might not have a `request` object. Limit could also be passed in as another parameter, defaulting to a 100 if it is not set.

Reply:
Good spot! I've updated the PR as per your suggestions which I actually think makes the code look neater too.